trade-finance-manager-ui/server/controllers/case/amendments/amendCoverEndDate.test.js

```diff
@@ -39,35 +44,89 @@ describe('POST postAmendCoverEndDate', () => {
 
   describe('TFM facility end date feature flag enabled', () => {
     beforeEach(() => {
-      jest.mocked(isTfmFacilityEndDateFeatureFlagEnabled).mockReturnValue(false);
+      jest.mocked(isTfmFacilityEndDateFeatureFlagEnabled).mockReturnValue(true);
     });
 
-    it('should redirect to is using facility end date page when no errors', async () => {
-      jest.mocked(isTfmFacilityEndDateFeatureFlagEnabled).mockReturnValue(true);
+    describe('GEF deal', () => {
+      it('should redirect to is using facility end date page when no errors', async () => {
+        api.getAmendmentById.mockResolvedValueOnce({ status: 200, data: MOCK_AMENDMENT_COVERENDDATE_CHANGE });
+        api.getFacility.mockResolvedValueOnce(gefFacility);
+        api.updateAmendment.mockResolvedValueOnce({ status: 200 });
+
+        const req = {
+          params: {
+            _id: MOCK_AMENDMENT_COVERENDDATE_CHANGE.dealId,
+            amendmentId: MOCK_AMENDMENT_COVERENDDATE_CHANGE.amendmentId,
+            facilityId: MOCK_AMENDMENT_COVERENDDATE_CHANGE.facilityId,
+          },
+          body: {
+            'cover-end-date-day': '12',
+            'cover-end-date-month': '8',
+            'cover-end-date-year': '2030',
+          },
+          session,
+        };
+
+        await postAmendCoverEndDate(req, res);
+
+        expect(res.redirect).toHaveBeenCalledWith(
+          `/case/${MOCK_AMENDMENT_COVERENDDATE_CHANGE.dealId}/facility/${MOCK_AMENDMENT_COVERENDDATE_CHANGE.facilityId}/amendment/${MOCK_AMENDMENT_COVERENDDATE_CHANGE.amendmentId}/is-using-facility-end-date`,
```

Comment:
can the different parts of this be dereferenced please? eg {dealId} = ... for each of these.  It is unreadable currently because it's so long

Reply:
Sure, have changed